src/System.Management.Automation/DscSupport/JsonCimDSCParser.cs

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.IO;
+using System.Management.Automation;
+using System.Security;
+
+namespace Microsoft.PowerShell.DesiredStateConfiguration.Internal.Json
+{
+    /// <summary>
+    /// Class that does high level Cim schema parsing.
+    /// </summary>
+    internal class CimDSCParser
+    {
+        private readonly JsonDeserializer _json_deserializer;
+        
+        internal CimDSCParser()
+        {
+            _json_deserializer = JsonDeserializer.Create();
+        }
+
+        internal IEnumerable<PSObject> ParseSchemaJson(string filePath, bool useNewRunspace = false)
+        {
+            try
+            {
+                string json = File.ReadAllText(filePath);
```

Comment:
This will use .NET's heuristics for encoding -- I assume that's ok?

Reply:
It was working fine so far.